### ISSUE TITLE
Column names in do not get quoted in shorthand methods

### ISSUE THREAD

**Issue by CodeDuck42** [doctrine/dbal#3635]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 2.9.2

#### Summary

Column names with reserved words do not get quoted.

\Doctrine\DBAL\Connection::gatherConditions();
\Doctrine\DBAL\Connection::insert();
\Doctrine\DBAL\Connection::update();

#### Current behaviour

In case of the insert it tries to run the following sql:

~~~sql
INSERT INTO test (id, order) VALUES (?,?);
~~~

#### How to reproduce

~~~sql
CREATE TABLE test (id integer, `order` integer);
~~~

~~~php
$connection->insert('test', [1, 1]);
~~~

#### Expected behaviour

Proper quoting of the column name from mysql

~~~sql
INSERT INTO table (id, `order`) VALUES (?,?);
~~~

**Comment by Ocramius** [comment]
This is correct and not currently fixable in `2.x` due to RDBMS case sensitivity issues and backwards compatibility.

Could be broken and fixed in 3.x, but then we'd have to know whether the input column names are quoted or not.

**Comment by morozov** [comment]
I don't think it's fixable on the DBAL side. I'd rather have ORM quote _all_ identifiers regardless of whether they are keywords or not.

**Comment by greg0ire** [comment]
@morozov can you please elaborate on this? We could have a configuration option turning it on on 3.0.x, that way people can adapt their code and pass unquoted identifiers everywhere.

**Comment by morozov** [comment]
My above comment is a bit outdated as of https://github.com/doctrine/dbal/issues/4772. The API consumer should be specific about whether the identifier should be quoted or not. The DBAL internally will quote all identifiers regardless of their case and them being a keyword.

> We could have a configuration option turning it on on 3.0.x, that way people can adapt their code and pass unquoted identifiers everywhere.

I don't think it's a good idea to have code semantics driven by runtime configuration. If that's implemented, you won't be able to implement code that is agnostic to the DBAL version or the settings. See `mbstring` overriding `strlen()` for example.
